src/main/java/sg/ncl/testbed_interface/DataResource.java

```diff
@@ -14,5 +14,44 @@
 
     private Long id;
     private String uri;
+    private String maliciousFlag;
+    private boolean malicious;
+    private boolean scanned;
+
+
+    // override the malicious flag method
+    // to display the list of dropdown selection for admin to update malicious status of a data resource
+    // Note: values must be equivalent to those option values defined in the dropdown list under admin_data_resources_update.html
+    public String getMaliciousFlag() {
+        if (malicious & scanned) {
+            maliciousFlag = "true";
+        } else if (!malicious & scanned) {
+            maliciousFlag = "false";
+        } else {
+            maliciousFlag = "not_scanned";
+        }
+        return maliciousFlag;
+    }
+
+    // override the malicious flag method
+    // to update the respective malicious and scanned flag
+    // Note: values must be equivalent to those define in the dropdown list under admin_data_resources_update.html
+    public void setMaliciousFlag(String flag) {
+        switch (flag) {
+            case "true":
+                malicious = true;
+                scanned = true;
+                break;
+            case "false":
+                malicious = false;
+                scanned = true;
+                break;
+            default:
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Move this default to the end of the switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)